src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -1964,14 +1981,15 @@ static int do_list_mapped_devices(const std::string &format, bool pretty_format)
       f->dump_string("pool", cfg.poolname);
       f->dump_string("namespace", cfg.nsname);
       f->dump_string("image", cfg.imgname);
-      f->dump_string("snap", cfg.snapname);
+      f->dump_string("snap", cfg.snapid.empty() ? cfg.snapname : "@" + cfg.snapid);
       f->dump_string("device", cfg.devpath);
       f->dump_string("cookie", cfg.cookie);
       f->close_section();
     } else {
       should_print = true;
       if (cfg.snapname.empty()) {
         cfg.snapname = "-";
+        cfg.snapname = cfg.snapid.empty() ? cfg.snapname : "@" + cfg.snapid;
```

Comment:
Assigning `cfg.snapname` twice in a row and potentially with `@<snapid>` doesn't read well.  I would do this instead:
```
      tbl << cfg.pid << cfg.poolname << cfg.nsname << cfg.imgname
          << (cfg.snapid.empty() ? cfg.snapname : "@" + cfg.snapid)
          << cfg.devpath << cfg.cookie << TextTable::endrow;
```